src/python.rs

```diff
@@ -54,10 +64,34 @@ struct PythonParser<'j> {
     tape: Tape,
     recursion_limit: u8,
     allow_inf_nan: bool,
+    allow_partial: bool,
 }
 
 impl<'j> PythonParser<'j> {
     fn py_take_value<StringCache: StringMaybeCache>(&mut self, py: Python, peek: Peek) -> JsonResult<PyObject> {
+        macro_rules! tri {
+            ($result:expr, $partial_value:expr) => {
+                match $result {
+                    Ok(k) => k,
+                    Err(e) => {
+                        return if self.allow_partial {
+                            match e.error_type {
+                                JsonErrorType::EofWhileParsingList
+                                | JsonErrorType::EofWhileParsingObject
+                                | JsonErrorType::EofWhileParsingString
+                                | JsonErrorType::EofWhileParsingValue
+                                | JsonErrorType::ExpectedListCommaOrEnd
+                                | JsonErrorType::ExpectedObjectCommaOrEnd => Ok($partial_value.to_object(py)),
```

Comment:
What's the motivation for these last two error codes being allowed here?

Reply:
they can be caused by the string ending unexpectedly, I guess we could split the errors, then use the same message so the output still matches serde?

Reply:
Happy either way, just wanted to understand these cases further. 👍